File: app/src/main/java/team/triplog/presentation/util/TextCountAnimation.java
package team.triplog.presentation.util;

import android.view.animation.Animation;
import android.view.animation.Transformation;
import android.widget.TextView;

import java.text.DecimalFormat;

public class TextCountAnimation extends Animation {
    TextView textView;
    int from = 0;
    int to;

    public TextCountAnimation(TextView textView, int to) {
        this.textView = textView;
        this.to = to;
    }

    DecimalFormat integerCommaFormat = new DecimalFormat("###,###");

    @Override
    public void applyTransformation(float interpolatedTime, Transformation t) {
        super.applyTransformation(interpolatedTime, t);
        setDuration(1000);
        textView.setText(integerCommaFormat.format((int) (from + ((to - from) * (double) interpolatedTime))));
    }
}

File: app/src/main/java/team/triplog/presentation/util/TextCountAnimation.kt
package team.triplog.presentation.util

import android.view.animation.Animation
import android.view.animation.Transformation
import android.widget.TextView
import java.text.DecimalFormat


/**
 * @author  mjkim
 * @since   2021.05.02
 */
class TextCountAnimation(var textView: TextView, var to: Int) : Animation() {
    var from = 0
    var integerCommaFormat = DecimalFormat("###,###")

    public override fun applyTransformation(interpolatedTime: Float, t: Transformation) {
        super.applyTransformation(interpolatedTime, t)
        duration = 1000
        textView.text = integerCommaFormat.format(
            (from + (to - from) * interpolatedTime.toDouble())
        )
    }
}